src/huggingface_hub/utils/endpoint_helpers.py

```diff
@@ -0,0 +1,306 @@
+
+        languages (:obj:`str` or :class:`List`, `optional`):
+            A string representing a two-character language to filter datasets by on the hub
+            Example usage:
+
+                >>> from huggingface_hub import DatasetFilter
+                >>> new_filter = DatasetFilter(language="en")
+
+        multilinguality (:obj:`str` or :class:`List`, `optional`):
+            A string representing a filter for datasets that contain multiple languages
+            Example usage:
+
+                >>> from huggingface_hub import DatasetFilter
+                >>> new_filter = DatasetFilter(multilinguality="yes")
+
+        size_categories (:obj:`str` or :class:`List`, `optional`):
+            A string or list of strings that can be used to identify datasets on the hub
+            by the size of the dataset such as `100K<n<1M` or `1M<n<10M`
+            Example usage:
+
+                >>> from huggingface_hub import DatasetFilter
+                >>> new_filter = DatasetFilter(size_categories="100K<n<1M")
+
+        task_categories (:obj:`str` or :class:`List`, `optional`):
+            A string or list of strings that can be used to identify datasets on the hub
+            by the designed task, such as `audio_classification` or `named_entity_recognition`
+            Example usage:
+
+                >>> from huggingface_hub import DatasetFilter
+                >>> new_filter = DatasetFilter(author="facebook")
+
+        task_ids (:obj:`str` or :class:`List`, `optional`):
+            A string or list of strings that can be used to identify datasets on the hub
+            by the specific task such as `speech_emotion_recognition` or `paraphrase`
+            Example usage:
+
+                >>> from huggingface_hub import DatasetFilter
+                >>> new_filter = DatasetFilter(task_ids="paraphrase")
+    """
+
+    author: str = None
+    benchmark: Union[str, List[str]] = None
+    dataset_name: str = None
+    language_creators: Union[str, List[str]] = None
+    languages: Union[str, List[str]] = None
+    multilinguality: Union[str, List[str]] = None
+    size_categories: Union[str, List[str]] = None
+    task_categories: Union[str, List[str]] = None
+    task_ids: Union[str, List[str]] = None
+
+
+@dataclass
+class ModelFilter:
+    """A class that converts human-readable model search parameters into ones compatible with
+    the REST API. For all parameters capitalization does not matter.
+
+    Args:
+        author (:obj:`str`, `optional`):
+            A string that can be used to identify models on the hub
+            by the original uploader (author or organization), such as `facebook` or `huggingface`
+            Example usage:
+
+                >>> from huggingface_hub import Filter
+                >>> new_filter = ModelFilter(author_or_organization="facebook")
+
+         framework (:obj:`str` or :class:`List`, `optional`):
```

Comment:
should this be library instead? That way we are consistent with `ModelSearchArguments` and the filters at left in https://huggingface.co/models

WDYT?

Reply:
Makes sense with me, will change. Nice catch